### PR TITLE
Make goto-def and renaming work for table and stream row type

### LINKED ISSUE #34801: Go-to definition and renaming doesn't work for table's row type parameter
**Description:**
$title

Rename_table_type_param_Peek 2022-01-24 14-26

Source code:
```ballerina
type Person record {|
    readonly int id;
    string name;
|};

type Student record {|
    *Person;
    int examYear?;
|};

function doThis() returns table<Student> {
    table<Person> tHuman = table [{id: 1, name: "Dulaj"}];

    table<Student> tStudent = tHuman.map(function(Person human) returns Student {
        return {
            id: human.id,
            name: human.name
        };
    });
    return tStudent;
}

```
Try renaming `table<Person<cursor>>`
**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose

Fixes #34801

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/references/ReferencesUtil.java [hunk 1]
```diff
@@ -120,7 +122,20 @@ public static Optional<Symbol> getSymbolAtCursor(PositionedOperationContext cont
...
         }
 
+        Document document = srcFile.get();
         Position position = context.getCursorPosition();
+        TextRange range = TextRange.from(
+                document.textDocument().textPositionFrom(PositionUtil.getLinePosition(position)), 0);
+        NonTerminalNode nonTerminalNode = ((ModulePartNode) document.syntaxTree().rootNode()).findNode(range);
+        SyntaxKind parentKind = nonTerminalNode.parent().kind();
+
+        if (parentKind == SyntaxKind.TYPE_PARAMETER || parentKind == SyntaxKind.STREAM_TYPE_PARAMS) {
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/references/ReferencesUtil.java:132]: Shall we include a comment mentioning the reason for searching the symbol at col-1

**Review by malinthar — CHANGES_REQUESTED** [review]
Changes LGTM! Let's add a comment as requested

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/references/ReferencesUtil.java (hunk 1 above)
Comment (KavinduZoysa) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/references/ReferencesUtil.java:132]: Added a comment

**Review by malinthar — APPROVED** [review]
LGTM!

**Review by mindula — APPROVED** [review]

**Pull request merged**